jujugui/static/gui/src/app/app.js

```diff
@@ -854,6 +858,46 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Callback handler for the api loginWithMacaroon method which hanadles
+      the "redirection required" error message.
+
+      @method _apiLoginHandler
+      @param {Object} api The API that the user is attempting to log into.
+      @param {String} err The login error message, if any.
+      @param {Object} data The login request response data, if any.
+    */
+    _apiLoginHandler: function(api, err, data) {
+      const errorNotify = function(err) {
+        this.db.notifications.add({
+          title: 'Unable to log into model',
+          message: `unable to log into model: ${err}`,
+          level: 'error'
+        });
+      };
+
+      if (err === 'authentication failed: redirection required') {
```

Comment:
I'd create a isRedirectError(err) function here, which encapsulates the error checking logic. String checking is already a weak approach, but it's the best we can do, but at least an external function will be more likely reused and if the message change there will be a single place affected.